auto_install/install.sh

```diff
@@ -2411,28 +2610,35 @@ confLogging() {
 	# Pre-create rsyslog/logrotate config directories if missing, to assure logs are handled as expected when those are installed at a later time
 	$SUDO mkdir -p /etc/{rsyslog,logrotate}.d
 
-	echo "if \$programname == 'ovpn-server' then /var/log/openvpn.log
-if \$programname == 'ovpn-server' then stop" | $SUDO tee /etc/rsyslog.d/30-openvpn.conf > /dev/null
-
-	echo "/var/log/openvpn.log
-{
-	rotate 4
-	weekly
-	missingok
-	notifempty
-	compress
-	delaycompress
-	sharedscripts
-	postrotate
-		invoke-rc.d rsyslog rotate >/dev/null 2>&1 || true
-	endscript
-}" | $SUDO tee /etc/logrotate.d/openvpn > /dev/null
+	{
+		echo "if \$programname == 'ovpn-server' then /var/log/openvpn.log"
+		echo "if \$programname == 'ovpn-server' then stop"
+	} | "${SUDO}" tee /etc/rsyslog.d/30-openvpn.conf > /dev/null
+
+	{
+		echo '/var/log/openvpn.log'
+		echo '{'
+		echo $'\t' 'rotate 4'
+		echo $'\t' 'weekly'
+		echo $'\t' 'missingok'
+		echo $'\t' 'notifempty'
+		echo $'\t' 'compress'
+		echo $'\t' 'delaycompress'
+		echo $'\t' 'sharedscripts'
+		echo $'\t' 'postrotate'
+		echo $'\t' $'\t' 'invoke-rc.d rsyslog rotate >/dev/null 2> /dev/stdout || true'
+		echo $'\t' 'endscript'
+		echo '}'
+	} | "${SUDO}" tee /etc/logrotate.d/openvpn > /dev/null
 
 	# Restart the logging service
-	case ${PLAT} in
-		Debian|Raspbian|Ubuntu)
+	case "${PLAT}" in
+		Debian | Raspbian | Ubuntu)
 			$SUDO systemctl -q is-active rsyslog.service && $SUDO systemctl restart rsyslog.service
 		;;
+		Alpine)
+			"${SUDO}" rc-service -s rsyslog restart
+		;;
```

Comment:
I suggest aligning this, so that `rsyslog` is only restarted when it is installed. All changes above in `installOpenVPN` can then be reverted, which, if I'm not mistaken, currently break the OpenVPN installation anyway, isn't it?

Reply:
To do as you say the entire use of `rsyslog` should be changed, as it is assumed to be installed (missing a forced installation step).
This is what prompted me to install it.

Reply:
I confirm that failure to install `rsyslog` stops the installer from running (as it should be since it is assumed to be present)

Reply:
Hmm, on DietPi we don't install `rsyslog` along with PiVPN anymore and never faced any issues with it. Why do you think it is expected? A config is installed, but it can make sense to do so just in case `rsyslog` is or will be installed at a later time.

> I confirm that failure to install `rsyslog` stops the installer from running

Since you force the service restart here. The the code above for Debian etc where this is only attempted if the service is actually running. Doing so on Alpine as well (I guess there is a `rc-service` equivalent to `systemctl -q is-active`) should work pretty fine without installing `rsyslog`.

Reply:
```
Usage: rc-service [options] [-i] <service> <cmd>...
   or: rc-service [options] -e <service>
   or: rc-service [options] -l
   or: rc-service [options] -r <service>

Options: [ cdDe:ilr:INsSZChqVv ]
  -d, --debug                       set xtrace when running the command
  -D, --nodeps                      ignore dependencies
  -e, --exists <arg>                tests if the service exists or not
  -c, --ifcrashed                   if the service is crashed run the command
  -i, --ifexists                    if the service exists run the command
  -I, --ifinactive                  if the service is inactive run the command
  -N, --ifnotstarted                if the service is not started run the command
  -s, --ifstarted                   if the service is started run the command
  -S, --ifstopped                   if the service is stopped run the command
  -l, --list                        list all available services
  -r, --resolve <arg>               resolve the service name to an init script
  -Z, --dry-run                     dry run (show what would happen)
  -h, --help                        Display this help output
  -C, --nocolor                     Disable color output
  -V, --version                     Display software version
  -v, --verbose                     Run verbosely
  -q, --quiet                       Run quietly (repeat to suppress errors)
```

So `rc-service -s rsyslog restart` works only if `rsyslog` is already started, instead of `rc-service -N rsyslog start` works only if `rsyslog` is installed but not started yet

Reply:
I could change the code from
```
rc-service -s rsyslog restart
rc-service -N rsyslog start
```
to
```
rc-service -s rsyslog restart || true
rc-service -N rsyslog start || true
```